### DISCUSSION TITLE
Modules implementation vs Builder usage for slash commands

### DISCUSSION THREAD

**Discussion by LastImmort** [discord-net/Discord.Net#2498] [Q&A]
What kind of implementation is better and more flexible for slash commands, with module or with builder?
It kind of confusing cause guide for slash commands point to builder implementation and handler. In the same time it could be done with modules and attributes in '.net core controller'  style, but looks like not all feature are available there.

**Comment by csmir** [comment]
The module implementation greatly reduces the amount of boilerplate a developer needs to implement to create, receive, resolve and handle any commands. In general, the modular implementation has far more advantages.
